api/src/main/java/org/phenotips/mendelianSearch/internal/DefaultPatientViewFactory.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.mendelianSearch.internal;
+
+import org.phenotips.data.Disorder;
+import org.phenotips.data.Feature;
+import org.phenotips.data.Patient;
+import org.phenotips.data.PatientData;
+import org.phenotips.data.PatientRepository;
+import org.phenotips.data.permissions.AccessLevel;
+import org.phenotips.data.permissions.PatientAccess;
+import org.phenotips.data.permissions.PermissionsManager;
+import org.phenotips.data.permissions.internal.visibility.HiddenVisibility;
+import org.phenotips.mendelianSearch.PatientView;
+import org.phenotips.mendelianSearch.PatientViewFactory;
+
+import org.xwiki.component.annotation.Component;
+
+import java.util.ArrayList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.ga4gh.GAVariant;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+
+/**
+ * Default implementation of {@link PatientViewFactory}.
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultPatientViewFactory implements PatientViewFactory
+{
+    private static final String UNDISCLOSED_MARKER = "?";
+
+    private static final String SOLVED_STRING = "solved";
+
+    @Inject
+    private PermissionsManager pm;
+
+    @Inject
+    @Named("view")
+    private AccessLevel viewAccess;
+
+    @Inject
+    private PatientRepository pr;
+
+    @Inject
+    private Provider<XWikiContext> xcontext;
+
+    @Override
+    public PatientView createPatientView(String id, List<GAVariant> variants, Double phenotypeScore,
+        MendelianSearchRequest request)
+    {
+        PatientView view;
+        Patient patient = this.pr.getPatientById(id);
+        if (patient == null) {
+            patient = this.pr.getPatientByExternalId(id);
+        }
+        if ((patient == null) || (variants == null) || (phenotypeScore == null)) {
+            return null;
+        }
+        boolean hasAccess = this.hasPatientAccess(patient);
```

Comment:
since you dont use the function anywhere else, you can probably inline it, or at least describe the access level you're restricting to here.